test/button/button-subtle.visual-diff.js

```diff
@@ -66,4 +59,14 @@ describe('d2l-button-subtle', function() {
 		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
 	});
 
+	const focus = async(page, selector) => {
+		return page.evaluate((selector) => {
+			return new Promise((resolve) => {
+				const elem = document.querySelector(selector);
+				elem.shadowRoot.querySelector('button').addEventListener('transitionend', resolve);
```

Comment:
Should we make `'button'` a parameter as well and move this to visual-diff helpers file for all tests?

Reply:
There certainly could be some value in that, and I was thinking about that a bunch last night.  

The reason I didn't was because desired element could be buried more than one shadow-root deep.  Nevertheless, that might cover a lot of our cases, at least for `core`.  

Then I started thinking that we could support a proprietary selector format such as: `#normal :: button`, and the helper could split on ` :: ` and query recursively query shadow-roots to locate the element.  I like that, except the proprietary part.

There's also the issue of the shadow-DOM changing which would break these types of tests, that that's true with the current impl too.

Reply:
Yeah I toyed with that too... it's not possible to pass elements between the nodejs and browser contexts.  It's possible to get a "handle" for an element that can be used in the nodejs context, and pass to certain functions, but I don't think it improves the consumer's code, which left me to this simple, albeit non-generic approach.  :(

I am sort of tempted to do the custom selector format.